software/robot_utilities.py

```diff
@@ -8,42 +8,65 @@
 
```

Comment:
Why did you write your own logging logic, instead of using Python provided logging functionality?

Reply:
Initially wasn't aware that Python has a logging module (but of course it has). Then just wanted to keep things simple and kept using the on we wrote.